contracts/adapters/UniswapAdapter.sol

```diff
@@ -0,0 +1,233 @@
+                return Exchange.Sushiswap;
+        }
+       require(false, 'invalid exchange');
+    }
+
+    function expectedWbtcOut(uint256 ethAmt, string memory exchangeName) public view returns (uint256){
+        if (keccak256(abi.encodePacked(exchangeName)) == uniswapHash){
+                address[] memory path = new address[](2);
+                path[0] = wethAddress;
+                path[1] = wbtcAddress;
+                uint256 wbtcOut = uniswapRouter.getAmountsOut(ethAmt, path)[1];
+                return wbtcOut;
+        }
+        if (keccak256(abi.encodePacked(exchangeName)) == sushiswapHash){
+                address[] memory path = new address[](2);
+                path[0] = wethAddress;
+                path[1] = wbtcAddress;
+                uint256 wbtcOut = sushiswapRouter.getAmountsOut(ethAmt, path)[1];
+                return wbtcOut;
+        }
+    }
+
+    function expectedDiggOut(uint256 wbtcAmt, string memory exchangeName) public view returns (uint256){
+        if (keccak256(abi.encodePacked(exchangeName)) == uniswapHash){
+                (uint112 reserve_amt,,) = IUniswapV2Pair(wbtcDiggUniswap).getReserves();
+                uint256 trade_amt = getSwapAmt(reserve_amt,wbtcAmt);
+                address[] memory path = new address[](2);
+                path[0] = wbtcAddress;
+                path[1] = diggAddress;
+                uint256 diggOut = uniswapRouter.getAmountsOut(trade_amt, path)[1];
+                return diggOut;
+        }
+        else if (keccak256(abi.encodePacked(exchangeName)) == sushiswapHash){
+                (uint112 reserve_amt,,) = IUniswapV2Pair(wbtcDiggSushiswap).getReserves();
+                uint256 trade_amt = getSwapAmt(reserve_amt,wbtcAmt);
+                address[] memory path = new address[](2);
+                path[0] = wbtcAddress;
+                path[1] = diggAddress;
+                uint256 diggOut = uniswapRouter.getAmountsOut(trade_amt, path)[1];
+                return diggOut;
+        }
+    }
+
+    function convertEthToToken( uint256 inputAmount, address addr,uint256 amountOutMin, Exchange exchange) internal {
+        IUniswapV2Router02 router = exchange == Exchange.Uniswap ? uniswapRouter : sushiswapRouter;
+        _convertEthToToken(inputAmount, addr, amountOutMin, router);
+    }
+
+    function convertTokenToToken(address addr1, address addr2, uint256 amount, uint256 amountOutMin, Exchange exchange) internal {
+        IUniswapV2Router02 router = exchange == Exchange.Uniswap ? uniswapRouter : sushiswapRouter;
+        _convertTokenToToken(addr1, addr2, amount, amountOutMin, router);
+    }
+
+    function addLiquidity(address token1,address token2,uint256 amount1,uint256 amount2,Exchange exchange) internal {
+        IUniswapV2Router02 router = exchange == Exchange.Uniswap ? uniswapRouter : sushiswapRouter;
+        _addLiquidity(token1, token2, amount1, amount2, router);

```

Comment:
Return the LP share amount so we can use it later

`return _addLiquidity(token1, token2, amount1, amount2, router);`